rclpy/rclpy/executors.py

```diff
@@ -387,15 +416,19 @@ def check_argument_type(callback_func, target_type):
 
                 # User might change the Timer.callback function signature at runtime,
                 # so it needs to check the signature every time.
-                arg_name = check_argument_type(tmr.callback, target_type=TimerInfo)
-                prefilled_arg = {arg_name: timer_info}
+                if tmr.callback:
+                    arg_name = check_argument_type(tmr.callback, target_type=TimerInfo)
```

Comment:
these code area is not directly related to type checking, but refactoring makes sense to me. thanks!